### ISSUE TITLE
!! torch.nn.functional.conv2d GPU cuda bug

### ISSUE THREAD

**Issue by chenhao1umbc** [pytorch/pytorch#37264]
## 🐛 Bug

`torch.nn.functional.conv2d` is **1000 times slower** with `print` using GPU

## To Reproduce
```
import torch, time
import torch.nn.functional as Func

s = torch.rand(300,48,1,200).cuda()
d = torch.rand(48,1, 256, 3).cuda()

t = time.time()
r = Func.conv2d(s, d, padding=(255, 1), groups=48)
# print(r.view(-1)[0])

print(time.time() - t)
```
Please compare with
```
import torch, time
import torch.nn.functional as Func

s = torch.rand(300,48,1,200).cuda()
d = torch.rand(48,1, 256, 3).cuda()

t = time.time()
r = Func.conv2d(s, d, padding=(255, 1), groups=48)
print(r.view(-1)[0])

print(time.time() - t)
```

## Expected behavior
first result
`0.00022411346435546875`

second result
```
tensor(0.3493, device='cuda:0')
3.597046136856079
```

The second code section should have comparable speed but it more than 1000 times slower!
instead of `print(r.view(-1)[0])`I have tried other codes like:
`rr = Func.conv2d(s[:,1].unsqueeze(1),d[0].unsqueeze(0), padding=(255, 1))`
It is also very slow. The slowness is not always happening.

## Environment
I am using google Colab Pro
```
 - PyTorch Version: 1.4 & 1.5 ( I also try 1.5)
 - OS (e.g., Linux): Unbuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: P100
 - Any other relevant information:

**Comment by fgvbrt** [comment]
It's not a but,  gpus ops are async here is relevant section:
https://pytorch.org/docs/stable/notes/cuda.html#asynchronous-execution

you can either set env variable or  `CUDA_LAUNCH_BLOCKING=1` or call ` torch.cuda.synchronize()` before measuring time according to this documentation

**Comment by chenhao1umbc** [comment]
OK, I see. Thanks.
